python/metatensor-operations/metatensor/operations/_dispatch.py

```diff
@@ -381,6 +381,8 @@ def mask(array, axis: int, mask):
     if isinstance(array, TorchTensor):
         return torch.index_select(array, dim=axis, index=indices)
     elif isinstance(array, np.ndarray):
+        if isinstance(indices, TorchTensor):
```

Comment:
I don't understand how this happens. I can see us indexing torch values with numpy indices (using classes from metatensor-core), but I don't see how the reverse happens here.

Reply:
I think that, with slice, we're at some point indexing both labels and the data tensor with the same index tensor. The index tensor is a torch.Tensor, while the labels are a numpy array

Reply:
ok, that makes sense.